main.go

```diff
@@ -54,89 +56,164 @@ package main
+	ra_tls_verify_callback_der_f = C.dlsym(ra_tls_verify_lib, ra_tls_verify_callback_der_sym)
 	if ra_tls_verify_callback_der_f == nil {
-		fmt.Errorf("error resolving %q function", ra_tls_verify_callback_der_sym)
-		return
+		panic(fmt.Errorf("error resolving %q function", ra_tls_verify_callback_der_name))
 	}
 
-	ra_tls_set_measurement_callback_sym := C.CString("ra_tls_set_measurement_callback")
+	ra_tls_set_measurement_callback := "ra_tls_set_measurement_callback"
+	ra_tls_set_measurement_callback_sym := C.CString(ra_tls_set_measurement_callback)
 	defer C.free(unsafe.Pointer(ra_tls_set_measurement_callback_sym))
-	ra_tls_set_measurement_callback_f := C.dlsym(ra_tls_verify_lib, ra_tls_set_measurement_callback_sym)
+	ra_tls_set_measurement_callback_f = C.dlsym(ra_tls_verify_lib, ra_tls_set_measurement_callback_sym)
 	if ra_tls_set_measurement_callback_f == nil {
-		fmt.Errorf("error resolving %q function", ra_tls_set_measurement_callback_sym)
-		return
+		panic(fmt.Errorf("error resolving %q function", ra_tls_set_measurement_callback))
 	}
+}
 
-	C.ra_tls_set_measurement_callback_wrapper(ra_tls_set_measurement_callback_f)
+func set_measurement_verification_args(mrenclave, mrsigner, isv_prod_id, isv_svn []byte) {
+	if len(mrenclave) == 0 {
+		C.set_g_verify_mrenclave(false)
+	} else {
+		C.set_g_verify_mrenclave(true)
+		C.set_g_expected_mrenclave(unsafe.Pointer(&mrenclave[0]), C.size_t(len(mrenclave)))
+	}
 
-	certFile := "tls/tlscert.der"
-	certDER, err := os.ReadFile(certFile)
+	if len(mrsigner) == 0 {
+		C.set_g_verify_mrsigner(false)
+	} else {
+		C.set_g_verify_mrsigner(true)
+		C.set_g_expected_mrsigner(unsafe.Pointer(&mrsigner[0]), C.size_t(len(mrsigner)))
+	}
 
-	if err != nil {
+	if len(isv_prod_id) == 0 {
+		C.set_g_verify_isv_prod_id(false)
+	} else {
+		C.set_g_verify_isv_prod_id(true)
+		C.set_g_expected_isv_prod_id(unsafe.Pointer(&isv_prod_id[0]), C.size_t(len(isv_prod_id)))
+	}
+
+	if len(isv_svn) == 0 {
+		C.set_g_verify_isv_svn(false)
+	} else {
+		C.set_g_verify_isv_svn(true)
+		C.set_g_expected_isv_svn(unsafe.Pointer(&isv_svn[0]), C.size_t(len(isv_svn)))
 	}
+}
+
+func RATLSVerifyDer(certDER, mrenclave, mrsigner, isv_prod_id, isv_svn []byte) error {
+	// set verify callback
+	C.ra_tls_set_measurement_callback_wrapper(ra_tls_set_measurement_callback_f)
+
+	// check for null for each measurement verification
+	set_measurement_verification_args(mrenclave, mrsigner, isv_prod_id, isv_svn)
 
 	cert_size := C.size_t(len(certDER))
 	certDER_sym := C.CBytes(certDER)
 	defer C.free(unsafe.Pointer(certDER_sym))
-	ret := C.ra_tls_verify_callback_der_wrapper(ra_tls_verify_callback_der_f, (*C.uchar)(certDER_sym), cert_size)
 
-	// convert der cert to C objects
-	// copy attestation consumables to static c objects
-	// call measurement callback
+	ret := C.ra_tls_verify_callback_der_wrapper(ra_tls_verify_callback_der_f, (*C.uchar)(certDER_sym), cert_size)
 
 	fmt.Println("Success: ", ret)
+	if ret != 0 {
```

Comment:
Return more verbose error message, for return values you can check: https://github.com/gramineproject/gramine/blob/master/tools/sgx/ra-tls/ra_tls_verify_dcap.c#L98

Unfortunately [the verification result is abstracted away](https://github.com/gramineproject/gramine/blob/master/tools/sgx/ra-tls/ra_tls_verify_dcap.c#L165-L185). We might wan't to work around this at some point.